src/main/docker/.env

```
@@ -0,0 +1,469 @@
+ARTEMIS_CONTINUOUSINTEGRATION_VCSAPPLICATIONLINKNAME="LS1 Bitbucket Server"
+# Do we need an empty commit for new exercises/repositories in order for the CI to register the repo
+ARTEMIS_CONTINUOUSINTEGRATION_EMPTYCOMMITNECESSARY=true
+# Hash/key of the ci-token, equivalent e.g. to the ci-token in version-control
+# Some CI systems, like Jenkins, offer a specific token that gets checked against any incoming notifications
+# from a VCS trying to trigger a build plan. Only if the notification request contains the correct token, the plan
+# is triggered. This can be seen as an alternative to sending an authenticated request to a REST API and then
+# triggering the plan.
+# In the case of Artemis, this is only really needed for the Jenkins + GitLab setup, since the GitLab plugin in
+# Jenkins only allows triggering the Jenkins jobs using such a token. Furthermore, in this case, the value of the
+# hudson.util.Secret is stored in the build plan, so you also have to specify this encrypted string here and NOT the actual token value itself!
+# You can get this by GETting any job.xml for a job with an activated GitLab step and your token value of choice.
+ARTEMIS_CONTINUOUSINTEGRATION_SECRETPUSHTOKEN="<token hash>"
+# Key of the saved credentials for the VCS service
+# Bamboo: not needed
+# Jenkins: You have to specify the key from the credentials page in Jenkins under which the user and
+#          password for the VCS are stored
+ARTEMIS_CONTINUOUSINTEGRATION_VCSCREDENTIALS="<credentials key>"
+# Key of the credentials for the Artemis notification token
+# Bamboo: not needed
+# Jenkins: You have to specify the key from the credentials page in Jenkins under which the authentication-token is stored
+ARTEMIS_CONTINUOUSINTEGRATION_ARTEMISAUTHENTICATIONTOKENKEY="<credentials key>"
+# The actual value of the notification token to check against in Artemis. This is the token that gets send with
+# every request the CI system makes to Artemis containing a new result after a build.
+# Bamboo: The token value you use for the Server Notification Plugin
+# Jenkins: The token value you use for the Server Notification Plugin and is stored under the notification-token credential above
+ARTEMIS_CONTINUOUSINTEGRATION_ARTEMISAUTHENTICATIONTOKENVALUE=<token>
+# Does cancel jenkins builds after 30 minutes to remove build that get stuck
+ARTEMIS_CONTINUOUSINTEGRATION_BUILDTIMEOUT=30
+ARTEMIS_CONTINUOUSINTEGRATION_BUILD_IMAGES_JAVA="ls1tum/artemis-maven-template:java17-8"
+ARTEMIS_CONTINUOUSINTEGRATION_BUILD_IMAGES_KOTLIN="ls1tum/artemis-maven-template:java17-8"
+ARTEMIS_CONTINUOUSINTEGRATION_BUILD_IMAGES_EMPTY="ls1tum/artemis-maven-template:java17-8"
+ARTEMIS_CONTINUOUSINTEGRATION_BUILD_IMAGES_PYTHON="ls1tum/artemis-python-docker:latest"
+ARTEMIS_CONTINUOUSINTEGRATION_BUILD_IMAGES_C="ls1tum/artemis-c-docker:latest"
+ARTEMIS_CONTINUOUSINTEGRATION_BUILD_IMAGES_C_FACT="sharingcodeability/fact:latest"
```

```suggestion
ARTEMIS_CONTINUOUSINTEGRATION_BUILD_IMAGES_CFACT="sharingcodeability/fact:latest"
```
With the underscore it would represent the key `artemis.continuousintegration.build.images.c.fact` in the YAML.

---

This causes an issue in https://github.com/ls1intum/Artemis/blob/f891dcf5f31c00a09a0b61d0c18445841c8747d7/src/main/java/de/tum/in/www1/artemis/service/connectors/ProgrammingLanguageConfiguration.java#L47
- When loading the config from YAML, the +1 is needed due to the additional underscore.
- When loading the config from ENV, it is not needed as the key is not allowed to contain the underscore.

---

Now already using the config values from your [PR 5596](https://github.com/ls1intum/Artemis/pull/5596). Gotta wait until that is merged and then merge the changes in here.